sagemaker-python-sdk/mxnet_gluon_mnist/mnist.py

```diff
@@ -53,6 +53,19 @@ def train(channel_input_dirs, hyperparameters, hosts, num_gpus, **kwargs):
     metric = mx.metric.Accuracy()
     loss = gluon.loss.SoftmaxCrossEntropyLoss()
 
+    # shard the training data in case we are doing
+    # distributed training.
+    if len(hosts) > 1:
+        train_data = [x for x in train_data]
+        shard_size = len(train_data) / len(hosts)
```

Comment:
These don't work when py_version is set to 'py3'.  Also, they leave a small amount of data out in py2.  Not sure that's a huge issue though since len(hosts) is typically going to be pretty small.